Various tweaks and improvements to the config and database management.

NatKarmios opened the pull request.

NatKarmios committed:
Removed setup_utils.py

NatKarmios committed:
Added config validation, moved constants to inner class

NatKarmios committed:
Revamped models.py and aryasorm.py to handle config injection

EXTREME reflection!

NatKarmios committed:
Added dependancies to requirements.txt

NatKarmios committed:
Renamed aryasorm.py to aryas_orm.py, setup no longer requires a command.

NatKarmios committed:
Fixed spelling mistake in _parse_config() docstring

NatKarmios committed:
Tweaked type hinting in aryas_orm.py and formatter.py

The type hinting syntax used caused PyCharm to give some warnings, this syntax will avoid such errors.

tom-sherman commented:
* Remove this stuff: `<editor-fold desc="config">`, `# noinspection PyClassHasNoInit` etc
* Change to single quote strings
* No need to return BaseModel or DiscordModel  on the models class, remove this

NatKarmios committed:
Removed editor-fold comments

NatKarmios committed:
Changed double-quoted strings to single-quoted.

NatKarmios committed:
Removed BaseModel and DiscordModel from Models class

NatKarmios committed:
Removed inspection-suppressing comment from models.py

NatKarmios committed:
Moved env from Config to Constants

NatKarmios committed:
Updated reference to env in models.py

NatKarmios committed:
Removed unneccesary Tuple import from models.py

NatKarmios committed:
Set SQLite DB to use aryas.db

NatKarmios committed:
Fixed accidental syntax error

NatKarmios committed:
Dump config errors to a file instead of printing

This removed the need for the pyyaml package; it has been removed from the dependancies.

NatKarmios committed:
Removed Config type hint from _Constants constructor

This caused a syntax error, because we all love circular dependancies.

NatKarmios committed:
Generate a default config file if not found

NatKarmios committed:
Added mod_log_channel_name and message_sleep_time to config schema

NatKarmios committed:
Fixed TypeError from misplaced reference

NatKarmios committed:
Merge branch 'master' into cfg-tweaks
